src/python/pants/binaries/binary_tool.py

```diff
@@ -75,17 +75,18 @@ def _get_archiver(self):
 
     return create_archiver(self.archive_type)
 
-  def url_generator(self):
+  def get_external_url_generator(self):
     """Override and return an instance of BinaryToolUrlGenerator to download from those urls.
 
     If this method returns None, urls to download the tool will be constructed from
     --binaries-baseurls. Otherwise, generate_urls() will be invoked on the result with the requested
     version and host platform.
 
-    If the bootstrap option --force-baseurls is set, the result of this method will be ignored. In
-    general, implementors of BinaryTool subclasses should try to ensure (within reason) that the
-    rest of their code can handle the result of downloading from different urls if url_generator()
-    is overridden yet again by a subclass, or if it is ignored with --force-baseurls.
+    If the bootstrap option --allow-external-binary-tool-downloads is False, the result of this
+    method will be ignored. In general, implementors of BinaryTool subclasses should try to ensure
```

Comment:
I'm not sure what this means? In what way would a BinaryTool subclass be affected by the download location, and how would an author mitigate that in code?

Reply:
Nit: Also, I think the word is spelled "implementer".